src/Extension/Challenge.cs

```diff
@@ -29,10 +29,10 @@ public Challenge(IReadOnlyList<EditableCode> content, Lesson lesson = null)
             Lesson = lesson;
         }
 
-        public async Task Evaluate(string submissionCode = null, IEnumerable<KernelEvent> events = null)
+        public async Task Evaluate(string submittedCode = null, IEnumerable<KernelEvent> events = null)
         {
             CurrentEvaluation = new Evaluation();
```

Comment:
this should not happen, CurrentEvaluation is history.peek()?.Evaluation

Reply:
updated